Allwinner: Bump u-boot to 2023.07-rc4 and ATF to 2.9.0

viraniac opened the pull request:
# Description

Bump U-boot to 2023.07-rc4, ATF to 2.9.0. As U-boot 2023.01+ requires scp.bin for H5, H6 and A64 platform, enabled crust with minimal config for the boards belonging to these platforms. Bigtreetech CB1 is kept back at v2023.04 as latest version requires some dram configurations that I was not comfortable with. It builds fine with v2023.04.

Jira reference number [AR-1753]
# How Has This Been Tested?

- [X] Built uboot for all of the allwinner boards
- [X] Booted on Nanopiduo2 and checked that crust continues to work fine

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules


[AR-1753]: https://armbian.atlassian.net/browse/AR-1753?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

viraniac commented:
@igorpecovnik As you haven't merged the 2023.01 merge request yet, I have updated it with 2023.04. hence requesting the review again.

schwar3kat reviewed (commented):
Orange Pi Zero Plus boots. 
Reboot works. 
Ethernet works. 
WiFi works.

As with previous builds suspend does not work.
Crust produces mixed/interesting results:
If systemctl suspend is called (which doesn't actually suspend) at any time before powerdown, then power drawn is slightly higher than without crust 0.18 Amps (vs 0.16) on powerdown.
If systemctl suspend has not been not called then on powerdown power drawn is 0.08 Amps.

On a fresh build, I commented out the "CRUSTCONFIG" line in the board config and Armbian build terminated with error  "binman: Filename 'scp.bin' not found in input path (.,.,./board/sunxi,arch/arm/dts)"

viraniac commented:
> As with previous builds suspend does not work.
> Crust produces mixed/interesting results:
> If systemctl suspend is called (which doesn't actually suspend) at any time before powerdown, then power drawn is slightly higher than without crust 0.18 Amps (vs 0.16) on powerdown.
> If systemctl suspend has not been called then on powerdown power drawn is 0.08 Amps.

@schwar3kat Could you please share the serial output before kernel starts and dmesg output? I am trying to findout why its not working for H5, but I haven't received my H5 board yet and hence am working blind here. I have found some kernel patches and will be sharing more images with you for testing purpose

> On a fresh build, I commented out the "CRUSTCONFIG" line in the board config and Armbian build terminated with error "binman: Filename 'scp.bin' not found in input path (.,.,./board/sunxi,arch/arm/dts)" I expected to be able to build without crust.

Thats stock u-boot behavior now. I can find out what change is causing it and revert the same if needed. But I am not sure if that is a good idea.

igorpecovnik commented:
> Thats stock u-boot behavior now. 

Yes, removing crust config is not enogh, one need to switch boot branch to last version that can work without crust, 22.10 IIRC ... for possible exception.

viraniac commented:
@schwar3kat Just found out, Its possible to build without crust even with 2023.07-rc4. You have to set the UBOOT_TARGET_MAP to following in the board config

```
UBOOT_TARGET_MAP="BINMAN_ALLOW_MISSING=1;;u-boot-sunxi-with-spl.bin"
```

@igorpecovnik Should I make this as default behavior and remove crustconfig from board files, or should we let individual maintainers to decide if they want to operate without crust?

schwar3kat commented:
> @schwar3kat Could you please share the serial output before kernel starts and dmesg and journalctl output? I am trying to find out why its not working for H5, but I haven't received my H5 board yet and hence am working blind here. ~I ~

Logs as requested.
[Orangepizeroplus serial U-Boot SPL 2023.07-rc4-armbian.log](https://github.com/armbian/build/files/11860772/Orangepizeroplus.serial.U-Boot.SPL.2023.07-rc4-armbian.log)
[Orangepizeroplus journalctl U-Boot SPL 2023.07-rc4-armbian.log](https://github.com/armbian/build/files/11860773/Orangepizeroplus.journalctl.U-Boot.SPL.2023.07-rc4-armbian.log)
[Orangepizeroplus dmesg U-Boot SPL 2023.07-rc4-armbian.log](https://github.com/armbian/build/files/11860774/Orangepizeroplus.dmesg.U-Boot.SPL.2023.07-rc4-armbian.log)

schwar3kat commented:
> UBOOT_TARGET_MAP="BINMAN_ALLOW_MISSING=1;;u-boot-sunxi-with-spl.bin"

**This is my  personal opinion, so feel free to ignore:**  
make UBOOT_TARGET_MAP="BINMAN_ALLOW_MISSING=1;;u-boot-sunxi-with-spl.bin" a default if CRUSTCONFIG is not found.
Allow board individual boards to bypass crust if required, at least until we have confidence that there are no issues.

CRUSTCONFIG in the board file would be useful for using alternate crust defconfig for debug and extra features like CIR.

viraniac commented:
> Logs as requested.
> [Orangepizeroplus serial U-Boot SPL 2023.07-rc4-armbian.log](https://github.com/armbian/build/files/11860772/Orangepizeroplus.serial.U-Boot.SPL.2023.07-rc4-armbian.log)
> [Orangepizeroplus journalctl U-Boot SPL 2023.07-rc4-armbian.log](https://github.com/armbian/build/files/11860773/Orangepizeroplus.journalctl.U-Boot.SPL.2023.07-rc4-armbian.log)
> [Orangepizeroplus dmesg U-Boot SPL 2023.07-rc4-armbian.log](https://github.com/armbian/build/files/11860774/Orangepizeroplus.dmesg.U-Boot.SPL.2023.07-rc4-armbian.log)

Thanks for sharing the logs. I can see in the u-boot log that crust was started and also from dmesg and journalctl that it was detected by Linux as well. I noticed linux detected rtc, but also gave some error while accessing it. So its possible that sun6i-rtc module might be buggy for H5 boards and thats why suspend does not work? I am not sure however if you tried to make the system go to sleep before capturing the log. Could you please run `systemctl suspend` and once the command completes then capture the log of journalctl. Also could you please share the output of `cat /sys/power/state` file.

viraniac commented:
> make UBOOT_TARGET_MAP="BINMAN_ALLOW_MISSING=1;;u-boot-sunxi-with-spl.bin" a default

Yes, I was thinking the same. I will make this default for sunxi64 so that maintainers can just delete CRUSTCONFIG to disable crust for their boards. I will however leave the crust config that I set for each board as it is for now until Igor shares his opinion about whether we want crust to be opt in or opt out.

AGM1968 commented:
sun6i-rtc module might be buggy

H616 has problems here which I've fixed... so very possible there are problems.

viraniac commented:
Changes in the latest version. Added BINMAN_ALLOW_MISSING=1 to the UBOOT_TARGET_MAP for sunxi64 so that if a maintainer wants to disable crust, they can do so just by deleting CRUSTCONFIG from the board configuration file. Also moved the change of setting CRUSTCONFIG from 2023.04 commit to 2023.01 commit as 2023.01 is the version that started enforcing presence of firmware blobs

schwar3kat commented:
> Could you please run `systemctl suspend` and once the command completes then capture the log of journalctl. Also could you please share the output of `cat /sys/power/state` file.

[Orangepizeroplus journalctl suspend U-Boot SPL 2023.07-rc4-armbian.log](https://github.com/armbian/build/files/11861410/Orangepizeroplus.journalctl.suspend.U-Boot.SPL.2023.07-rc4-armbian.log)

`cat /sys/power/state`
`freeze mem`

This board doesn't have a hardware rtc, so that might cause an error,

viraniac commented:
> Looks like "rtl8189fs mmc1:0001:1: PM: failed to suspend async: error -38" is preventing sleep. This is probably not crust related. It looks like a misbehaving WiFi module.

Just for the testing purpose, could you please try blacklisting the module so it won't load on boot and then test how crust behaves on your board. I know this will mean that wifi on the board won't work, but I just want to see what if the system properly suspends without it.



> This board doesn't have a hardware rtc, so that might cause an error,

H5 CPU does have a builtin rtc, that is why sun6i-rtc module is getting loaded. Once you blacklist the wifi module, you should be able to use rtcwake command to put system to sleep

igorpecovnik reviewed (approved):
I am fine if we merge this at present state.

schwar3kat commented:
> Once you blacklist the wifi module, you should be able to use rtcwake command to put system to sleep

With WiFi blacklisted (blacklist cfg80211 and blacklist 8189fs)

rtcwake -m mem -s 30 works - 0.00 Amps.
systemctl suspend works - 0.04 Amps.
